src/Build/Resources/Constants.cs

```diff
@@ -365,6 +365,23 @@ private static void InitializeAvailableMethods()
                         availableStaticMethods.TryAdd("Microsoft.Build.Utilities.ToolLocationHelper", new Tuple<string, Type>("Microsoft.Build.Utilities.ToolLocationHelper, Microsoft.Build.Utilities.Core, Version=" + MSBuildConstants.CurrentAssemblyVersion + ", Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", null));
                         availableStaticMethods.TryAdd("System.Runtime.InteropServices.RuntimeInformation", runtimeInformationType);
                         availableStaticMethods.TryAdd("System.Runtime.InteropServices.OSPlatform", osPlatformType);
+#if NET5_0_OR_GREATER
+                        var operatingSystemType = new Tuple<string, Type>(null, typeof(OperatingSystem));
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsOSPlatform", operatingSystemType);
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsOSPlatformVersionAtLeast", operatingSystemType);
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsLinux", operatingSystemType);
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsFreeBSD", operatingSystemType);
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsFreeBSDVersionAtLeast", operatingSystemType);
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsMacOS", operatingSystemType);
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsMacOSVersionAtLeast", operatingSystemType);
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsWindows", operatingSystemType);
+                        availableStaticMethods.TryAdd("System.OperatingSystem::IsWindowsVersionAtLeast", operatingSystemType);
```

Comment:
@akoeplinger Which way do you want it?

> I'd keep just these methods ...

or

> I don't think we need to limit to specific methods ...

because the methods we expose from `System.OperatingSystem` need to be matched by methods on `Microsoft.Build.Framework.OperatingSystem`.

Reply:
> they shouldn't need to be matched, we can have more methods available on .NET 5 than on .NET Framework.

Working on the change.